src/main.cpp

```diff
@@ -2,209 +2,13 @@
-  // the mesh
-
-  // Disable some fancy rendering
-  auto rs = sceneCamera->getRenderSettings();
-
-  rs->screenSpaceReflections.enabled = false;
-  rs->ambientOcclusion.enabled = false;
-  rs->enableIndirectLighting = true;
-  rs->enableFXAA = false;
-  rs->enableHDR = false;
-  rs->enableTonemapping = false;
-
-  sceneCamera->setRenderSettings(rs);
-
-  HSceneObject worldSO = {};
-
-  // Import a Gothic ZEN
-  if (BsZenLib::HasCachedZEN(zenFile))
-  {
-    HPrefab worldPrefab = BsZenLib::LoadCachedZEN(zenFile);
-
-    if (!worldPrefab)
-    {
-      gDebug().logError("Failed to load cached ZEN: " + zenFile);
-      return -1;
-    }
-
-    worldPrefab.blockUntilLoaded();
-
-    worldSO = worldPrefab->instantiate();
-  }
-  else
-  {
-    worldSO =
-        BsZenLib::ImportAndCacheZEN(zenFile.c_str(), REGoth::gVirtualFileSystem().getFileIndex());
-
-    if (!worldSO)
-    {
-      gDebug().logError("Failed to load uncached ZEN: " + zenFile);
-      return -1;
-    }
-  }
-
-  HSceneObject startSO = worldSO->findChild("zCVobStartpoint:zCVob");
-
-  Vector3 startPosition = startSO ? startSO->getTransform().getPosition() : Vector3(bs::BsZero);
-  Vector3 startLookAt = startSO ? startSO->getTransform().getForward() : Vector3(1.0, 0.0, 0.0);
-
-  // Position the camera
-  sceneCameraSO->setParent(playerSO);
-
-  HCharacterController charController = playerSO->addComponent<CCharacterController>();
-  playerSO->addComponent<FPSWalker>();
-
-  // Make the character about 1.8m high, with 0.4m radius (controller represents a capsule)
-  charController->setHeight(1.0f);  // + 0.4 * 2 radius = 1.8m height
-  charController->setRadius(0.4f);
-
-  HFPSCamera fpsCamera = sceneCameraSO->addComponent<FPSCamera>();
-  fpsCamera->setCharacter(playerSO);
-
-  sceneCameraSO->setPosition(Vector3(0.0f, 1.8f * 0.5f - 0.1f, 2.0));
-
-  playerSO->setPosition(startPosition);
-  // playerSO->lookAt(startLookAt);
-
-  // setupInputConfig();
-
-  // Add GUI
-  HSceneObject guiSO = SceneObject::create("GUI");
-
-  float guiScale = 1.0f;
-  guiSO->setScale(Vector3(guiScale, guiScale, guiScale));
-  HGUIWidget gui = guiSO->addComponent<CGUIWidget>(sceneCamera);
-  gui->setSkin(BuiltinResources::instance().getGUISkin());
-
-  GUIPanel* mainPanel = gui->getPanel();
-
-  // Add draw distance slider
-  GUISlider* drawDistanceSlider = mainPanel->addNewElement<GUISliderHorz>();
-
-  drawDistanceSlider->setRange(1.0f, 100.0f);
-  drawDistanceSlider->setPosition(5, 5);
-  drawDistanceSlider->setWidth(200);
-
-  drawDistanceSlider->onChanged.connect([&](float percent) {
-    float value = drawDistanceSlider->getValue();
-
-    auto rs = sceneCamera->getRenderSettings();
-
-    rs->cullDistance = value;
-    gDebug().logDebug("Set CullDistance to: " + toString(value));
-
-    sceneCamera->setRenderSettings(rs);
-  });
-
-  regoth.run();
-
-  return 0;
+  return ::REGoth::runEngine(engine);
 }
```

Comment:
@ataulien I also removed the legacy code right here. I think it would be the best, if we make `REGoth` the main executable to actually just play. This means that this main.cpp is very small, as it just has to instantiate the plain defaults to get started.

All other executables are pretty much development, single purpose execs which modify the default engine to make a certain trait better testable/debuggable. Please let me know if this is okay or if I misunderstood anything.

Reply:
I have some code laying around defining a `REGothGame` class inherited from `REGothEngine` which sets up normal gameplay. That way we can have a simple `main.cpp` and also define test executables extending the game class if we want to.

Reply:
Is it not feasible to try to make `REGothEngine` the default near-production implementation and override certain aspects of the engine if-need-be for tests/debugs? 

But you're saying that this change is legitimate, right? I mean, the legacy code is in Git anyway, so why keep it in the code base, no?